sympy/core/basic.py

```diff
@@ -57,7 +114,7 @@ def _old_compare(x: type, y: type) -> int:
     return (i1 > i2) - (i1 < i2)
 
 
-class Basic(Printable, metaclass=ManagedProperties):
+class Basic(_BasicBase, Printable):
```

Comment:
What's the advantage of having `Basic(_BasicBase, Printable)` over having `_BasicBase` subclass `Printable`?

Reply:
There's no difference because they don't share any methods. _BasicBase doesn't have anything to do with printing, so it makes more sense to not do this.

Reply:
I will remove `_BasicBase` and use a `_prepare_class_assumptions` function as mentioned below.

Reply:
That's better. I don't like the idea of having another class in the mro to confuse people (even if it does begin with an underscore).

Reply:
I've done this.